code/game/jobs/job/job.dm

```diff
@@ -114,17 +117,28 @@
 
 	return max(0, minimal_player_age - C.player_age)
 
+/// Returns true if the character has a disability the selected job doesn't allow
 /datum/job/proc/barred_by_disability(client/C)
-	if(!C)
-		return 0
-	if(disabilities_allowed)
-		return 0
-	var/list/prohibited_disabilities = list(DISABILITY_FLAG_BLIND, DISABILITY_FLAG_DEAF, DISABILITY_FLAG_MUTE, DISABILITY_FLAG_DIZZY)
-	for(var/i = 1, i < prohibited_disabilities.len, i++)
-		var/this_disability = prohibited_disabilities[i]
-		if(C.prefs.active_character.disabilities & this_disability)
-			return 1
-	return 0
+	if(!C || !blacklisted_disabilities)
```

Comment:
`!blacklisted_disabilities` will never succeed since it's assigned an empty list (not null). Need to check length instead.
```suggestion
	if(!C || !length(blacklisted_disabilities))
```

Reply:
I even tested this in code-testing, then forgot to implement it 😄